src/Grid/Column.js

```diff
@@ -32,24 +34,26 @@ const getOffset = (offset, columns) => css`
 const StyledColumn = createComponent({
   name: 'Column',
   as: Flex,
-  style: p => {
-    const sizes = Object.keys(p.theme.grid.sizes);
-    const mediaQueries = sizes.filter(size => !!p[size]).map(
-      size =>
-        // if smallest breakpoint no query needed
-        size === head(sizes)
-          ? css`
-              ${getColumnWidth(p[size], p.theme.grid.columns)};
-              ${p[`${size}Offset`] && getOffset(p[`${size}Offset`], p.theme.grid.columns)};
-            `
-          : // all other breakpoints use a media query
-            css`
-              @media (min-width: ${p.theme.grid.sizes[size]}px) {
-                ${getColumnWidth(p[size], p.theme.grid.columns)};
-                ${p[`${size}Offset`] !== undefined && getOffset(p[`${size}Offset`], p.theme.grid.columns)};
-              }
-            `
-    );
+  style: ({ theme, order, ...props }) => {
+    const mediaQueries = Object.keys(theme.breakpoints).map(breakpoint => {
+      if (!props[breakpoint]) {
+        return null;
+      }
+
+      if (breakpoint === 'xs') {
+        return css`
+          ${getColumnWidth(props[breakpoint], theme.gridColumns)};
+          ${props[`${breakpoint}Offset`] && getOffset(props[`${breakpoint}Offset`], theme.gridColumns)};
+        `;
+      }
+
+      return css`
+        @media (min-width: ${theme.breakpoints[breakpoint]}) {
+          ${getColumnWidth(props[breakpoint], theme.gridColumns)};
+          ${props[`${breakpoint}Offset`] !== undefined && getOffset(props[`${breakpoint}Offset`], theme.gridColumns)};
+        }
+      `;
```

Comment:
Could we use breakpoint aliases to generate these media queries?

Reply:
I'd rather not touch the implementation in this PR